Formula/a/alpscore.rb

```diff
@@ -4,6 +4,7 @@ class Alpscore < Formula
   url "https://github.com/ALPSCore/ALPSCore/archive/refs/tags/v2.3.1.tar.gz"
   sha256 "384f25cd543ded1ac99fe8238db97a5d90d24e1bf83ca8085f494acdd12ed86c"
   license "GPL-2.0-only"
+  revision 1
```

Comment:
ALPSCore  is going to take extra work to build with our CMake-built HDF5 which behaves different from Autotools-built variant. Currently won't be detected correctly with: https://github.com/ALPSCore/ALPSCore/blob/v2.3.1/common/cmake/ALPSCommonModuleDefinitions.cmake#L166-L167

Reply:
It's odd to set `HDF5_NO_FIND_PACKAGE_CONFIG_FILE` to true, but I guess it's legit when you want to workaround cmake bugs or if you want to do something custom.

I tried defining HDF5_INCLUDE_DIRS and HDF5_LIBRARIES, the error message change from:
```
-- Unable to determine HDF5 C flags from HDF5 wrapper.
CMake Error at /usr/local/Cellar/cmake/3.30.3/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find HDF5 (missing: HDF5_LIBRARIES) (found suitable version
  "1.14.4.3", minimum required is "1.10.2")
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.30.3/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.30.3/share/cmake/Modules/FindHDF5.cmake:1028 (find_package_handle_standard_args)
  common/cmake/ALPSCommonModuleDefinitions.cmake:167 (find_package)
  hdf5/CMakeLists.txt:36 (add_hdf5)
```

to 


```
-- Unable to determine HDF5 C flags from HDF5 wrapper.
CMake Error at /usr/local/Cellar/cmake/3.30.3/share/cmake/Modules/FindHDF5.cmake:1084 (message):
  HDF5 was found, but a different variable was set which contains the
  location of the `hdf5::hdf5` library.
Call Stack (most recent call first):
  common/cmake/ALPSCommonModuleDefinitions.cmake:167 (find_package)
  alea/CMakeLists.txt:43 (add_hdf5)
```

Reply:
I do have a workaround in #183022